HeterogeneousCore/CUDACore/python/ProcessAcceleratorCUDA.py

```diff
@@ -0,0 +1,28 @@
+import FWCore.ParameterSet.Config as cms
+
+import os
+
+class ProcessAcceleratorCUDA(cms.ProcessAccelerator):
+    def __init__(self):
+        super(ProcessAcceleratorCUDA,self).__init__()
+        self._label = "gpu-nvidia"
+    def labels(self):
+        return [self._label]
+    def enabledLabels(self):
+        enabled = (os.system("cudaIsEnabled") == 0)
+        if enabled:
+            return self.labels()
+        else:
+            return []
+    def apply(self, process, accelerators):
+        if not hasattr(process, "CUDAService"):
+            from HeterogeneousCore.CUDAServices.CUDAService_cfi import CUDAService
+            process.add_(CUDAService)
+
+        if self._label in accelerators:
+            process.CUDAService.enabled = True
+            process.MessageLogger.CUDAService = cms.untracked.PSet()
```

Comment:
Maybe this should be set also of the CUDAService is disabled, in order to see the corresponding message in the log ?

Reply:
Good point, but at this point it is probably best to leave it for a subsequent PR (as I assume you're effectively suggesting with the `+heterogeneous`)

Reply:
sure

Reply:
Also, for the follow up PR, maybe this could be change to check if `process.MessageLogger.CUDAService` does not already exist, to avoid overwriting any values it may have.